lib/std/Build.zig

```diff
@@ -2065,22 +2163,17 @@ pub fn dependencyFromBuildZig(
 }
 
 fn userValuesAreSame(lhs: UserValue, rhs: UserValue) bool {
+    if (std.meta.activeTag(lhs) != rhs) return false;
```

Comment:
I don't like this function because it means that the syntax sucks so much people are using std.meta instead of basic syntax for unions. However, you can keep it here as a visible blight to encourage me to make it nicer to access the active tag of a union with zig syntax. I think removing `@unionTag` was a mistake. The motivation was since it could be implemented in the standard library, the language could be simplified. But I think that was bad reasoning.

Reply:
what's the reasoning for `==` and `!=` not working on unions directly?

Reply:
I'd guess ambiguity and consistency, it's not obvious whether the comparison interacts with the payload or not